### PR TITLE
Refactor into ProUserAccount & PartnerUserAccount

### PR DESCRIPTION
**Pull request by tazsingh**
**What kind of change does this PR introduce?**

In essence, a `Partner` is any type of company that does business with Honeywell/Resideo. It can be an insurance company, marketing, advertising, pros, etc.

Thus having a `PartnerAccount` that is meant to consolidate all of these various entities and their user accounts is potentially problematic.

This PR introduces a `ProUserAccount` to be specifically for Pro users and a `PartnerUserAccount` for any Partner. The intended model is as follows:

User Role | GraphQL Types
----------|----------------
Homeowner/Customer/End User | `UserAccount`
Pro Portal User/Installer | `UserAccount` & `PartnerUserAccount` &  `ProUserAccount`
Insurance/Marketing/Telecom | `UserAccount` & `PartnerUserAccount` & Future entity specific for their role

**Does this PR introduce a breaking change?**

Currently, no as we're not using this schema in production.

However, it is a change from the schema in Pro Portal /cc @shanewilson 

**What needs to be documented once your changes are merged?**

Hopefully, this is documented well enough within the schema files. If not, reviewers let me know your suggestions on how to improve!

### CODE REVIEW

File: src/services/ProUser/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  proUserAccount: ProUserAccount
+}
+
+"""
+A UserAccount extension for Pro Portal Users
+"""
+type ProUserAccount implements Node @key(fields: "id") {
+  id: ID!
+
+  employeeId: ID
```
Comment (tazsingh) [inline src/services/ProUser/schema.graphql]: I'm a little bit unsure if this should go here or on the `PartnerUserAccount`. Thoughts? /cc @chuckcarpenter

Comment (shanewilson) [reply src/services/ProUser/schema.graphql]: Pro and non-Pro users would both have `PartnerUserAccount` right? Might be better to put it there if they both have/need `employeeId`s.

Comment (tazsingh) [reply src/services/ProUser/schema.graphql]: Yeah that's right. A Pro is a Partner, but a Partner isn't strictly a Pro. Ok, I'll move `employeeId` onto the Partner. I agree that it makes more sense there 👍

Comment (chuckcarpenter) [reply src/services/ProUser/schema.graphql]: I agree, b/c I understood it as for internal use to the partner.

**Pull request merged**